account_invoice_production_lot/models/account_invoice.py

```diff
@@ -35,17 +36,38 @@ def _compute_prod_lots(self):
         for line in self:
             if not line.order_line_ids:
                 return
-            line.prod_lot_ids = self.mapped(
-                'order_line_ids.procurement_ids.move_ids.lot_ids')
+            line.prod_lot_ids = self.mapped('move_line_ids.lot_ids')
 
     @api.multi
     def _compute_line_lots(self):
         for line in self:
             if line.prod_lot_ids:
+                res = line.quantity_by_lot()
                 note = u'<ul>'
-                note += u' '.join([
-                    u'<li>S/N {0}</li>'.format(lot.name)
-                    for lot in line.prod_lot_ids
-                ])
+                lot_strings = []
+                for lot in line.prod_lot_ids:
+                    lot_string = u'<li>S/N %s%s</li>' % (
+                        lot.name, u' (%s)' % res[lot] if res.get(lot) else '')
+                    lot_strings.append(lot_string)
+                note += u' '.join(lot_strings)
                 note += u'</ul>'
                 line.lot_formatted_note = note
+
+    def quantity_by_lot(self):
+        move_ids = self.mapped('move_line_ids')
```

Comment:
Add one of `self.ensure_one` or `@api.multi`, your choice

Reply:
Done in https://github.com/OCA/account-invoice-reporting/pull/85/commits/b0f167deba50042fae1060a6e949173d717de247

Reply:
Since you chose `ensure_one`, the `mapped` is not needed anymore

Reply:
Done in https://github.com/OCA/account-invoice-reporting/pull/85/commits/526c171623466dc80b7d401b3c287352f937b897